.github/workflows/main.yml

```diff
@@ -0,0 +1,75 @@
+name: Test and Generate Docs
+
+on: pull_request
+
+jobs:
+  go-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
+      
+      - name: Set up Go
+        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed
+        with:
+          go-version-file: go.mod
+
+      - name: Install datadog-ci
+        run: npm install -g @datadog/datadog-ci
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y git python3-pip
+          go install github.com/jstemmer/go-junit-report@v1.0.0
+          go install github.com/kyoh86/richgo@v0.3.10
+          go install gotest.tools/gotestsum@latest
+          pip install pre-commit
+
+      - name: Run pre-commit hooks
+        run: pre-commit run -a golangci-lint
```

Comment:
Use the golangci-lint action instead, this also means you don't need to install :
```
    - uses: golangci/golangci-lint-action@aaa42aa0628b4ae2578232a66b541047968fac86 # v6.1.0
      with:
        version: v1.53.3
```

This also makes the running pre-commit redundant unless we add additional pre-commit hooks there.